eras/alonzo/test-suite/src/Test/Cardano/Ledger/Alonzo/AlonzoEraGen.hs

```diff
@@ -433,12 +426,26 @@ instance Mock c => EraGen (AlonzoEra c) where
         Nothing -> storageCost 0 pp script
       else storageCost 0 pp script
 
-  genEraDone pp tx =
+  -- For some reason, the EraGen generators occasionally generate an extra script witness.
+  --    There is some evidence that this arises because the script hash appears as the PolicyId
+  --    in a Value. But that is not been verified. Regardless of the cause, we can fix this by
+  --    discarding the trace. Note that this is failure to generate a "random" but valid
+  --    transaction. Discarding the trace adjust for this inadequacy in the generation process.
+  --    This only appears in the Alonzo era, so this "fix" is applied here, in the genEraDone
+  --    method of the EraGen class in the (AlonzoEra c) instance.
```

Comment:
maybe we could leave a `FIXME` or `TODO` to address the root cause? I'm happy having a discoverable discard over constant flaky tests breaking CI, but if/when we get into discard madness it's good to be able to find them... (though maybe searching for `myDiscard` will lead us to them).

Reply:
I will make an issue, referencing this PR and some notes.